giza/giza/operations/stage.py

```diff
@@ -416,9 +416,17 @@ def stage(self, root):
         if not os.path.isdir(root):
             raise NoSuchEdition(root)
 
+        # If a redirect is masking a file, we can run into an invalid 404
+        # when the redirect is deleted but the file isn't republished.
+        # If this is the case, warn and delete the redirect.
+        for src,dest in redirects.items():
+            src_path = os.path.join(root, src)
+            if os.path.basename(src_path) in os.listdir(os.path.dirname(src_path)):
+                LOGGER.warn('Redirect will mask file: %s', src)
```

Comment:
we might as well add to logging message that we're deleting the offending redirect.